### DISCUSSION TITLE
字体警告 Package fontspec Warning

### DISCUSSION THREAD

**Discussion by lewis-1231** [stone-zeng/fduthesis#346] [Q&A] [answered]
在我使用学位论文模板（2024年12月数学学院官网版，见https://math.fudan.edu.cn/30396/list.htm ）的时候，有以下三条warning：
1. Package fontspec Warning: Font "FandolSong-Regular" does not contain requested Script "CJK".
2. Package fontspec Warning: Font "FandolHei-Regular" does not contain requested Script "CJK".
3. Package fontspec Warning: Font "FandolKai-Regular" does not contain requested Script "CJK". 

查阅README文档时，文档建议在fdusetup的子选项style中添加命令: 
cjk-font = (\*<adobe(fandol)|founder|mac|sinotype|sourcehan|windows|none>\*).
我尝试了这个方法，结果会报错:
LaTeX Error: Key 'fdu/style/cjk-font' accepts only a fixed set of choices. 
我又尝试将其改为
cjk-font = fandol.
此时不会报错，但是那三条warning并没有消失．

请问有朋友会解决这个问题吗？

**Comment by SwitWu** [comment, marked as answer]
从 2024/5/11 开始，fontspec 宏包已经将此类 warning 降级为 info，见 https://github.com/latex3/fontspec/commit/046963860d7f8995aeab170152087a59d935c1ee

将你使用的 fontspec 宏包更新为此版本（或后续版本）即可。

**Reply by lewis-1231** [reply]
谢谢！

**Reply by tanukihee** [reply]
同时，`cjk-font = (*<adobe(fandol)|founder|mac|sinotype|sourcehan|windows|none>*)` 的意思是让你把这个选项写成 `cjk-font = adobe` 或 `cjk-font = fandol` 等等这样多个选项中选一个的样子，而不是把这一整行全粘上去…

**Reply by lewis-1231** [reply]
确实...
